### PR TITLE
[wasm] Fix System.Core test suite plus a bug in the icall code. First pass at fixing System test suite.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

**Comment by akoeplinger** [comment]
@monojenkins build

File: sdks/wasm/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-TOP=$(realpath $(CURDIR)/../..)
+TOP:=$(realpath $(CURDIR)/../..)
```
Comment (luhenry) [inline sdks/wasm/Makefile]: Why this change?

File: sdks/wasm/Makefile [hunk 2]
```diff
@@ -72,35 +72,27 @@ $(patsubst %,managed/%,$(BCL_ASSEMBLIES)): managed/%: $(WASM_BCL_DIR)/% | manage
 
 #FIXME change BCL sdk build to include nunitlite.dll in out/bcl/wasm/
 managed/nunitlite.dll: $(TOP)/mcs/class/lib/wasm/nunitlite.dll | managed/
-	cd $(TOP)/mcs/tools/nunit-lite && make PROFILE=wasm
+	cd $(TOP)/mcs/tools/nunit-lite && $(MAKE) PROFILE=wasm
```
Comment (luhenry) [inline sdks/wasm/Makefile]: Prefer `$(MAKE) -C $(TOP)/mcs/tools/nunit-lite` over `cd ... && $(MAKE)`

File: sdks/wasm/Makefile [hunk 3]
```diff
@@ -118,3 +110,5 @@ run-%: toolchain build
 clean:
 	
 package:
+
+FORCE:
```
Comment (luhenry) [inline sdks/wasm/Makefile]: what's that target for?

File: mono/mini/wasm_m2n_invoke.g.h [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /*
- * DON'T EDIT THIS FILE
- * This file was generated by m2n-gen.cs - use it instead.
- */
+* DON'T EDIT THIS FILE
```
Comment (luhenry) [inline mono/mini/wasm_m2n_invoke.g.h]: Line damage

File: sdks/wasm/Makefile (hunk 1 above)
Comment (kumpera) [reply sdks/wasm/Makefile]: Damage, let me remove it.

File: sdks/wasm/Makefile (hunk 2 above)
Comment (kumpera) [reply sdks/wasm/Makefile]: Fixed.

File: sdks/wasm/Makefile (hunk 3 above)
Comment (kumpera) [reply sdks/wasm/Makefile]: This is to make the targets that depend on it to be always reevaluated. IE, make them behave just like PHONY target.

The bcl test suite targets must be PHONY so we get proper dep tracking by the nested make invocation.
This makes the dev cycle nicer as I can edit the test suite and just type `make run-foo` and not have to clear or touch files.

The alternative to this trick is to rename those targets from actual files to proper PHONY targets. Which I find unfortunate, as the actual file being generated gets obscured.

Comment (luhenry) [reply sdks/wasm/Makefile]: Why don't you do `.PHONY: $(TOP)/mcs/class/lib/wasm/tests/wasm_corlib_test.dll`? That will force to reevaluate the `$(TOP)/mcs/class/lib/wasm/tests/wasm_corlib_test.dll` target, even if the file exists

**Comment by kumpera** [comment]
Closing this PR as https://github.com/mono/mono/pull/6488 includes all commits from this one.
